Neos.ContentRepository.Core/Classes/NodeType/NodeType.php

```diff
@@ -490,11 +493,34 @@ public function hasAutoCreatedChildNode(NodeName $nodeName): bool
      */
     public function getTypeOfAutoCreatedChildNode(NodeName $nodeName): ?NodeType
     {
+        $this->initialize();
         return isset($this->fullConfiguration['childNodes'][$nodeName->value]['type'])
             ? $this->nodeTypeManager->getNodeType($this->fullConfiguration['childNodes'][$nodeName->value]['type'])
             : null;
     }
 
+    /**
+     * Calculates the deterministic nodeAggregateIds for the auto-created childNodes.
+     */
+    public function tetheredNodeAggregateIds(NodeAggregateId $parentNodeAggregateId): NodeAggregateIdsByNodePaths
+    {
+        $nodeAggregateIdsByNodePaths = NodeAggregateIdsByNodePaths::createEmpty();
+        foreach ($this->getAutoCreatedChildNodes() as $childNodeName => $childNodeType) {
+            $nodeAggregateIdsByNodePaths = $nodeAggregateIdsByNodePaths->add(
+                $childNodePath = NodePath::fromString($childNodeName),
+                $childNodeAggregateId = NodeAggregateId::fromParentNodeAggregateIdAndNodeName($parentNodeAggregateId, NodeName::fromString($childNodeName))
+            );
+
+            // handle multiple levels of autocreated child nodes
+            foreach ($childNodeType->tetheredNodeAggregateIds($childNodeAggregateId)->getNodeAggregateIds() as $nestedChildNodeName => $nodeAggregateId) {
```

Comment:
howdy there, my name is re cursion.

Reply:
howdy there, I think there should be some kind of endless-recursion-check :)
I mean.. If you have something like
```yaml
'Some:NodeType':
  children:
    'foo':
      type: 'Some:NodeType'
```
it probably won't work anyways. But maybe we should detect those cases anyways to provide a better error message than "memory limit exceeded..."